### ISSUE TITLE
Can't translate OrderBy on a generic method

### ISSUE THREAD

**Issue by TanielianVB** [dotnet/efcore#20181]
Hi,
When ordering by a property defined on a interface witch restricts the generic type of the IQueryable I get the "could not be translated" error.

### Steps to reproduce
With the following table:
``` SQL
CREATE TABLE public."Things"
(
    "Guid" bigint NOT NULL,
    CONSTRAINT "Things_pkey" PRIMARY KEY ("Guid")
)
```
The following sample:
``` C#
public class EFCoreTestContext : DbContext
{
    public DbSet<Thing> Things { get; set; }

    public static readonly ILoggerFactory DebugLoggerFactory = LoggerFactory.Create(builder => { builder.AddFilter((category, level) => category == DbLoggerCategory.Database.Command.Name && level == LogLevel.Information).AddConsole(); });

    protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder) => optionsBuilder
        .UseLoggerFactory(DebugLoggerFactory)
        //.UseNpgsql("Host=localhost;Database=EFCoreTest;Username=test;Password=test")
        .UseSqlServer(@"Data Source=(localDB)\MSSQLLocalDB;Initial Catalog=EFCoreTest;Integrated Security=True");

    protected override void OnModelCreating(ModelBuilder modelBuilder)
    {
        modelBuilder
            .Entity<Thing>(eb =>
            {
                eb.ToTable("Things");
                eb.HasKey(e => e.Guid);
            });
    }
}

public class Thing
{
    public long Guid { get; set; }
}

public interface IDTO
{
    long ID { get; set; }
}

public class DTO : IDTO
{
    public long ID { get; set; }
}

public class ThingDTO : DTO { }

class Program
{
    static IQueryable<TEntity> OrderById<TEntity>(IQueryable<TEntity> query) where TEntity : IEntity => query.OrderBy(e => e.ID);

    static void Main(string[] args)
    {
        using (var db = new EFCoreTestContext())
        {
            var q = (from t in db.Things
                        select new ThingDTO
                        {
                            ID = t.Guid,
                        });

            var r = q.OrderBy(e => e.ID).ToList();

            var r1 = r.ToList();

            Console.WriteLine();

            try
            {
                var r2 = OrderById(q).ToList();
            }
            catch (Exception ex)
            {
                Console.ForegroundColor = ConsoleColor.Red;
                Console.WriteLine(ex.Message);
            }
        }

        Console.ReadKey();
    }
}
```
I'm getting this output:
``` SQL
--info
--: Microsoft.EntityFrameworkCore.Database.Command[20101]
      Executed DbCommand (4ms) [Parameters=[], CommandType='Text', CommandTimeout='30']
      SELECT t."Guid" AS "ID"
      FROM "Things" AS t
      ORDER BY t."Guid"
```
``` C#
//The LINQ expression 'DbSet<Thing>
    .OrderBy(t => ((IEntity)new ThingDTO{ ID = t.Guid }
    ).ID)//' could not be translated. Either rewrite the query in a form that can be translated, or switch to client evaluation explicitly by inserting a call to either AsEnumerable(), AsAsyncEnumerable(), ToList(), or ToListAsync(). See https://go.microsoft.com/fwlink/?linkid=2101038 for more information.
```
### Further technical details

EF Core version: 3.1.2
Database provider: Microsoft.EntityFrameworkCore.SqlServer or Npgsql.EntityFrameworkCore.PostgreSQL both 3.1.2
Target framework: (e.g. .NET Framework 4.7.2
Operating system: Windows
IDE: Visual Studio 2019 16.4.5

**Comment by smitpatel** [comment]
Duplicate of #20097

**Comment by TanielianVB** [comment]
@smitpatel , thank you.
Is it possible the tell me in which package version the commit 62b88f3e654527d15934bd2d125143afc013971e will be in?
Thanks in advance,

**Comment by smitpatel** [comment]
It is available in daily builds. It will be released as part of 5.0.0 preview3.
